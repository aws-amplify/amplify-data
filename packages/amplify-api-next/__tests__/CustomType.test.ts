import { expectTypeTestsToPassAsync } from 'jest-tsd';
import { a } from '../index';

// evaluates type defs in corresponding test-d.ts file
it('should not produce static type errors', async () => {
  await expectTypeTestsToPassAsync(__filename);
});

describe('CustomType transform', () => {
  test('Explicit CustomType', () => {
    const s = a.schema({
      Post: a.model({
        location: a.ref('Location'),
      }),
      Location: a.customType({
        lat: a.float(),
        long: a.float(),
      }),
    });

    const result = s.transform().schema;

    expect(result).toMatchSnapshot();
  });

  test('Explicit CustomType - required', () => {
    const s = a.schema({
      Post: a.model({
        location: a.ref('Location').required(),
      }),
      Location: a.customType({
        lat: a.float(),
        long: a.float(),
      }),
    });

    const result = s.transform().schema;

    expect(result).toMatchSnapshot();
  });

  test('Explicit CustomType - with auth', () => {
    const s = a.schema({
      Post: a.model({
        location: a.ref('Location').authorization([a.allow.owner()]),
      }),
      Location: a.customType({
        lat: a.float(),
        long: a.float(),
      }),
    });

    const result = s.transform().schema;

    expect(result).toMatchSnapshot();
  });

  test('Implicit CustomType', () => {
    const s = a.schema({
      Post: a.model({
        location: a.customType({
          lat: a.float(),
          long: a.float(),
        }),
      }),
    });

    const result = s.transform().schema;

    expect(result).toMatchSnapshot();
  });
});
