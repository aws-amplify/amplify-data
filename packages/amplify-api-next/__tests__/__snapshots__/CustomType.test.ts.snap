// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomType transform Explicit CustomType - required 1`] = `
"type Post @model 
{
  location: Location!
}

type Location 
{
  lat: Float
  long: Float
}"
`;

exports[`CustomType transform Explicit CustomType - with auth 1`] = `
"type Post @model 
{
  owner: String
  location: Location @auth(rules: [{allow: owner, ownerField: "owner"}])
}

type Location 
{
  lat: Float
  long: Float
}"
`;

exports[`CustomType transform Explicit CustomType 1`] = `
"type Post @model 
{
  location: Location
}

type Location 
{
  lat: Float
  long: Float
}"
`;

exports[`CustomType transform Implicit CustomType 1`] = `
"type Post @model 
{
  location: Location
}

type Location 
{
  lat: Float
  long: Float
}"
`;
